File: data-structures/java/src/main/java/std/data/struct/util/Stack.java
package std.data.struct.util;

import java.util.LinkedList;
import java.util.List;

public class Stack {

    private List<Object> list = new LinkedList();

    public void push(Object element) {
        list.add(element);
    }

    public Object pop() {
        return this.list.remove(this.list.size() - 1);
    }

    public Object peek() {
        return this.list.get(this.list.size() - 1);
    }

    public boolean isEmpty() {
        return list.isEmpty();
    }

    @Override
    public String toString() {
        return this.list.toString();
    }
}

File: data-structures/typescript/src/stack.ts
import { LinkedList } from "./linked-list";

export class Stack<Type> {
  private list = new LinkedList<Type>();

  constructor(firstElement?: Type) {
    if (firstElement) {
      this.push(firstElement);
    }
  }

  size(): number {
    return this.list.getSize();
  }

  isEmpty(): boolean {
    return this.size() === 0;
  }

  push(element: Type): void {
    this.list.add(element);
  }

  pop(): Type {
    if (this.isEmpty()) {
      throw new Error("Empty stack");
    }

    return this.list.removeLast();
  }

  peek(): Type {
    if (this.isEmpty()) {
      throw new Error("Empty stack");
    }

    return this.list.peekLast();
  }
}
